### PR TITLE
DEVX-2490: Adding session windows tutorial for KStreams

### PR DESCRIPTION
**Pull request by bbejeck**
### Description

Covers https://github.com/confluentinc/kafka-tutorials/issues/691

### Staging Docs

http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/DEVX-2490_add_kafka_streams_session_windows/
 http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/DEVX-2490_add_kafka_streams_session_windows/create-session-windows/kstreams.html

### CODE REVIEW

**Comment by gAmUssA** [comment]
This is awesome @bbejeck 
I will try to review this today

File: _includes/tutorials/session-windows/kstreams/code/src/main/java/io/confluent/developer/SessionWindow.java [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+            streams.cleanUp();
+            streams.start();
+            latch.await();
+        } catch (Throwable e) {
+            System.exit(1);
+        }
+        System.exit(0);
+    }
+
+    static class ClicksDataGenerator {
```
Comment (bbejeck) [inline _includes/tutorials/session-windows/kstreams/code/src/main/java/io/confluent/developer/SessionWindow.java:162]: I included an inner class to generate records for this tutorial.  Usually, I would use the data-gen connector but session windows require setting timestamps with enough time to enforce a new session starting, and this approach makes it easier to create separate "sessions".  I've noted in the tutorial the use of this class and that users shouldn't include this in their code.   
 
IMHO it's a better user experience to have the input topic populate with records automatically vs. having them go through the step themselves.

\cc @gAmUssA

File: _includes/tutorials/session-windows/kstreams/code/src/main/java/io/confluent/developer/SessionWindow.java [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+                sessionClicks.add(Clicks.newBuilder().setIp(keyOne).setUrl("/mauris/morbi/non.jpg").setTimestamp(instant.plusMillis(24000).toEpochMilli()).build());
+                sessionClicks.add(Clicks.newBuilder().setIp(keyOne).setUrl("/nullam/orci/pede/venenatis.json").setTimestamp(instant.plusMillis(38000).toEpochMilli()).build());
+
+                sessionClicks.add(Clicks.newBuilder().setIp(keyTwo).setUrl("/etiam/justo/etiam/pretium/iaculis.xml").setTimestamp(instant.plusMillis(10000).toEpochMilli()).build());
+                sessionClicks.add(Clicks.newBuilder().setIp(keyTwo).setUrl("/mauris/morbi/non.jpg").setTimestamp(instant.plusMillis(32000).toEpochMilli()).build());
+                sessionClicks.add(Clicks.newBuilder().setIp(keyTwo).setUrl("/nec/euismod/scelerisque/quam.xml").setTimestamp(instant.plusMillis(44000).toEpochMilli()).build());
+                sessionClicks.add(Clicks.newBuilder().setIp(keyTwo).setUrl("/nullam/orci/pede/venenatis.json").setTimestamp(instant.plusMillis(58000).toEpochMilli()).build());
+
+                Instant newSessionInstant = instant.plus(2,ChronoUnit.HOURS);
```
Comment (bbejeck) [inline _includes/tutorials/session-windows/kstreams/code/src/main/java/io/confluent/developer/SessionWindow.java:191]: This line is what forces a new session

File: _includes/tutorials/session-windows/kstreams/code/src/main/java/io/confluent/developer/SessionWindow.java (hunk 1 above)
Comment (gAmUssA) [reply _includes/tutorials/session-windows/kstreams/code/src/main/java/io/confluent/developer/SessionWindow.java:162]: i love it. IIUC this datagen is on-off, right? not continuous

File: _includes/tutorials/session-windows/kstreams/code/src/main/java/io/confluent/developer/SessionWindow.java (hunk 2 above)
Comment (gAmUssA) [reply _includes/tutorials/session-windows/kstreams/code/src/main/java/io/confluent/developer/SessionWindow.java:191]: nit - space is missing after `2, ChronoUnit.HOURS`

File: _includes/tutorials/session-windows/kstreams/code/src/main/java/io/confluent/developer/SessionWindow.java (hunk 1 above)
Comment (bbejeck) [reply _includes/tutorials/session-windows/kstreams/code/src/main/java/io/confluent/developer/SessionWindow.java:162]: > is on-off, right? not continuous

Yes, it produces 8 records then quits

File: _includes/tutorials/session-windows/kstreams/code/src/main/java/io/confluent/developer/SessionWindow.java (hunk 2 above)
Comment (bbejeck) [reply _includes/tutorials/session-windows/kstreams/code/src/main/java/io/confluent/developer/SessionWindow.java]: ack

**Review by gAmUssA — APPROVED** [review]
+1 LGTM

**Comment by bbejeck** [comment]
the test for the PR passed https://confluentinc.semaphoreci.com/jobs/e9886293-6781-4a64-bd0a-2ef6ab9faa61

**Pull request merged**